src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -203,10 +203,29 @@ void evaluate_impl(
       "e.g. evaluate<ti::a, ti::b>(L, R(ti::b, ti::a));, where R's first "
       "index has 2 spatial dimensions but L's second index has 3 spatial "
       "dimensions. Check RHS and LHS indices that use the same generic index.");
+  static_assert(Derived::height_relative_to_closest_tensor_leaf_in_subtree <
+                    std::numeric_limits<size_t>::max(),
+                "This either indicates that no Tensors were found in the RHS "
+                "TensorExpression or that the depth of the tree exceeded the "
+                "maximum size_t value. If there is indeed a Tensor in the RHS "
+                "expression, this indicates a flaw in the logic for the "
+                "derived TensorExpression types' member, "
+                "height_relative_to_closest_tensor_leaf_in_subtree.");
 
```

Comment:
Is there a `static_assert` somewhere already that the type `X` must be either `DataVector` or a fundamental? If not, this code certainly makes it a requirement. It should be noted in that assertion that the `DataVector` restriction can be generalized to other vector types (we have modal and complex(modal) vectors)

Reply:
No it isn't, and I think that is something that is long overdue, because there are several things besides this change that would break if someone used a different data type. I will add this assert